homeassistant/components/media_player/apple_tv.py

```diff
@@ -6,134 +6,179 @@
 """
 import asyncio
 import logging
-import hashlib
-
-import voluptuous as vol
+import os
 
 from homeassistant.core import callback
+from homeassistant.components.apple_tv import (
+    ATTR_ATV, ATTR_POWER, DATA_APPLE_TV)
 from homeassistant.components.media_player import (
     SUPPORT_NEXT_TRACK, SUPPORT_PAUSE, SUPPORT_PREVIOUS_TRACK, SUPPORT_SEEK,
     SUPPORT_STOP, SUPPORT_PLAY, SUPPORT_PLAY_MEDIA, SUPPORT_TURN_ON,
-    SUPPORT_TURN_OFF, MediaPlayerDevice, PLATFORM_SCHEMA, MEDIA_TYPE_MUSIC,
-    MEDIA_TYPE_VIDEO, MEDIA_TYPE_TVSHOW)
+    SUPPORT_TURN_OFF, MediaPlayerDevice, MEDIA_TYPE_MUSIC,
+    MEDIA_TYPE_VIDEO, MEDIA_TYPE_TVSHOW, DOMAIN, MEDIA_PLAYER_SCHEMA)
 from homeassistant.const import (
     STATE_IDLE, STATE_PAUSED, STATE_PLAYING, STATE_STANDBY, CONF_HOST,
-    STATE_OFF, CONF_NAME, EVENT_HOMEASSISTANT_STOP)
-from homeassistant.helpers.aiohttp_client import async_get_clientsession
-import homeassistant.helpers.config_validation as cv
+    STATE_OFF, CONF_NAME, EVENT_HOMEASSISTANT_STOP, ATTR_ENTITY_ID)
+from homeassistant.loader import get_component
+from homeassistant.config import load_yaml_config_file
 import homeassistant.util.dt as dt_util
 
 
-REQUIREMENTS = ['pyatv==0.2.1']
+DEPENDENCIES = ['apple_tv']
 
 _LOGGER = logging.getLogger(__name__)
 
-CONF_LOGIN_ID = 'login_id'
-CONF_START_OFF = 'start_off'
+SERVICE_AUTHENTICATE = 'apple_tv_authenticate'
+
+DATA_ENTITIES = 'data_apple_tv_entities'
+
+KEY_CONFIG = 'apple_tv_configuring'
 
-DEFAULT_NAME = 'Apple TV'
+NOTIFICATION_ID = 'apple_tv_notification'
+NOTIFICATION_TITLE = 'Apple TV Authentication'
 
-DATA_APPLE_TV = 'apple_tv'
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Required(CONF_HOST): cv.string,
-    vol.Required(CONF_LOGIN_ID): cv.string,
-    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-    vol.Optional(CONF_START_OFF, default=False): cv.boolean
-})
+def request_configuration(hass, config, atv, credentials):
```

Comment:
It probably can, but it depends on how we want to handle entity lookups in the service handler. Since entity ids passed are passed to apple_tv_authenticate (like with all services), the entity objects must be saved in the media_player platform (which is done now) but retrieved from the component. Not an implementation problem in any way, more a technical thing if it's "OK" to have the component-platform-coupling this way?